packages/react-dev-utils/WatchChangedNodeModulesPlugin.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// This Webpack plugin ensures that any change in node_modules package.json forces a project rebuild.
+
+'use strict';
+
+var fs = require('fs');
+
+var timeout;
+
+class WatchChangedNodeModulesPlugin {
+  constructor(nodeModulesPath) {
+    this.nodeModulesPath = nodeModulesPath;
+  }
+
+  apply(compiler) {
+    fs.watch(
```

Comment:
why not use `https://github.com/webpack/watchpack`? also i think watching the whole files in `node_modules` would be very expensive, can we just watch all the `package.json`s inside it? and ignore some folders that would be changed on runtime like `.cache`

Reply:
Hi, great to see you!
Webpack watch is current direction and changing [a RegExp](https://github.com/facebook/create-react-app/blob/5348d6eecf208a6298e48072314157804282002e/packages/react-scripts/config/webpackDevServer.config.js#L78-L80) used to ignore `node_modules`.

Reply:
Hi, great to see you too!
Ah, I see, that would be better! would the `ignored` setting accept a function? I imagine it would be easier to read than a regex with lots of condition. Reading the docs it only accept regex or anymatch, but in some place like loaders' test it accept a function like `(filePath) => filePath.endsWith('.js')`